server.go

```diff
@@ -72,19 +119,32 @@ func (s *Server) Handle(ctx context.Context, conn net.Conn) {
 	s.clientByID.Store(c.id, c)
 	defer func() {
 		s.clientByID.Delete(c.id)
+		close(c.monitorChan)
```

Comment:
Let's delete the channel from `monitorChans` and then close it, to be safe against the farfetched scenario of someone pushing on a closed channel.

We could even let the client close it on its `Close()` method, since `Client.Close` is the last thing that takes place before `Handle` returns.